repositories/duty.repository.ts

```diff
@@ -0,0 +1,51 @@
+import { UpdateFilter, Filter, ObjectId, Collection } from 'mongodb';
+
+import { getDutiesCollection } from '../db_connection';
+import { Duty } from '../services/schemas/duty.zschema';
+import { modifyFilterForArraySupport } from './modifiers';
+
+let collection: Readonly<Collection<Duty>>;
+
+(async () => {
```

Comment:
Top level await is possible in node > 16

Reply:
OK, is this a problem?

Reply:
Yes. put this in a function and then await the function. This is much less clean.
Also, don't do this. this will cause a side effect whenever you import the function.

Reply:
resolved